src/Features/CSharp/Portable/ReplaceDefaultLiteral/CSharpReplaceDefaultLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,150 @@
+            // This code fix addresses very specific compiler errors. It's unlikely there will be more than 1 of them at a time.
+            return null;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var syntaxRoot = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var token = syntaxRoot.FindToken(context.Span.Start);
+
+            if (token.Span == context.Span &&
+                token.IsKind(SyntaxKind.DefaultKeyword) &&
+                token.Parent.IsKind(SyntaxKind.DefaultLiteralExpression))
+            {
+                var defaultLiteral = (LiteralExpressionSyntax)token.Parent;
+                var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                var (newExpression, displayText) = GetReplacementExpressionAndText(
+                    context.Document, defaultLiteral, semanticModel, context.CancellationToken);
+
+                if (newExpression != null)
+                {
+                    context.RegisterCodeFix(
+                        new MyCodeAction(
+                            c => ReplaceAsync(context.Document, context.Span, newExpression, c),
+                            displayText),
+                        context.Diagnostics);
+                }
+            }
+        }
+
+        private static async Task<Document> ReplaceAsync(
+            Document document, TextSpan span, SyntaxNode newExpression, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var defaultToken = syntaxRoot.FindToken(span.Start);
+            var defaultLiteral = (LiteralExpressionSyntax)defaultToken.Parent;
+
+            var newRoot = syntaxRoot.ReplaceNode(defaultLiteral, newExpression.WithTriviaFrom(defaultLiteral));
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        private static (SyntaxNode newExpression, string displayText) GetReplacementExpressionAndText(
+            Document document,
+            LiteralExpressionSyntax defaultLiteral,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var type = semanticModel.GetTypeInfo(defaultLiteral, cancellationToken).ConvertedType;
+            if (type != null && type.TypeKind != TypeKind.Error)
+            {
+                if (IsFlagsEnum(type, semanticModel.Compilation) &&
+                    type.GetMembers("None").FirstOrDefault() is IFieldSymbol field && IsZero(field.ConstantValue))
+                {
+                    return GenerateMemberAccess("None");
+                }
+                else if (type.Equals(semanticModel.Compilation.GetTypeByMetadataName("System.Threading.CancellationToken")))
```

Comment:
nit: we will often do `typeof(System.Threading.CancellationToken).FullName` so you don't have to hardcode the string.